common/build/eslint-config-fluid/src/custom-rules/no-restricted-tags-imports.js

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+module.exports = {
+	meta: {
+		type: "problem",
+		docs: {
+			description:
+				"This rule restricts imports from internal or non-public APIs. This to prevent accidental dependencies on internal, unstable or undocumented parts of the codebase.",
+			category: "Best Practices",
+		},
+		fixable: "code",
+		schema: [
+			{
+				type: "object",
+				properties: {
+					tags: {
+						type: "array",
+						items: { type: "string" },
+						uniqueItems: true,
+					},
+					exceptions: {
+						type: "array",
+						items: { type: "string" },
+						uniqueItems: true,
+					},
+				},
+				additionalProperties: false,
+			},
+		],
+		messages: {
+			restrictedImport: "Import ",
+		},
+	},
+	create(context) {
+		const options = context.options[0] || {};
+		const restrictedTags = new Set(options.tags || []);
+		const exceptions = new Set(options.exceptions || []);
+		return {
+			ImportDeclaration(node) {
+				console.log(node.source);
+				const isException = exceptions.has(node.source.value);
+				if (isException) {
+					return; // Skip further checks for this import
+				}
+				node.specifiers.forEach((items) => {
+					const variable = context.getDeclaredVariables(items)[0];
+					const sourceCode = context.getSourceCode();
+					const comments = sourceCode.getAllComments();
+					// JSDocs comments are block comments, so we're only interested in those.
+					comments.forEach((comment) => {
+						if (comment.type !== "Block") {
+							return;
+						}
+						// Use a JSDoc parser to parse the comments for each imported item.
+						const jsDoc = require("doctrine").parse(comment.value, { unwrap: true });
```

Comment:
I left a separate recommendation to use a different library here, but as general feedback, we usually prefer require calls be at the top like import statements in typescript code.